src/propertiesdialog.cpp

```diff
@@ -370,19 +370,13 @@ void PropertiesDialog::apply()
 
     saveShortcuts();
```

Comment:
this causes a `Properties::Instance()->saveSettings();` [inside](https://github.com/lxqt/qterminal/pull/1146/files/ab8ccaf23aa6631e36fadef9e27221a784b3905b..8509f7e1a55f11bd194b197d6cff6fe78517431d#diff-fb415beef169a9cbd4c61db798927f7032fedfa56d3abf9174e85716f35e3d87R461) it, will it affect anything by doing that here? then a bit later below at line [397](https://github.com/lxqt/qterminal/pull/1146/files/ab8ccaf23aa6631e36fadef9e27221a784b3905b..8509f7e1a55f11bd194b197d6cff6fe78517431d#diff-fb415beef169a9cbd4c61db798927f7032fedfa56d3abf9174e85716f35e3d87R397) ? ie. a double save so quickly?

Reply:
Yes, I noticed it too but left it as it was. Will remove it before merging the PR.

IMO, The code structure of QTerminal is far from ideal — each class may suppose some things about another, there are redundant computations, nullity checks aren't sufficient, etc. So, in general, some sort of restructuring may be needed, but it should be done with the utmost care to prevent regressions. Perhaps a job for the future...